### PR TITLE
runtime: resolve shared home edge cases

### PR DESCRIPTION
**Pull request by zevorn**
## Summary

- resolve relative `OH_DSH_HOME` values before launching the macOS app
- keep migrations retryable when Windows link targets are unavailable
- cover both edge cases with launcher and migration regression tests

## Verification

- `node --test tests/cli.test.ts tests/data-root.test.ts`
- `pnpm typecheck`
- `pnpm test`
- `pnpm build`

## Context

These fixes address the final review findings from #52. They were reviewed in
stacked PR #55, then rebased onto `main` because #52 had already merged.

### CODE REVIEW

**Comment by zevorn** [comment]
@codex review

**Comment by zevorn** [comment]
@codex review

File: src/data-root.ts [hunk 1]
```diff
@@ -215,11 +222,16 @@ function copyDirectoryContents(
...
     if (options.exclude?.has(entry) === true) continue
-    copyEntry(join(source, entry), join(destination, entry), roots)
+    copied = copyEntry(
+      join(source, entry),
+      join(destination, entry),
+      roots,
+    ) && copied
   }
-  return true
+  return copied
```
[the remark this answers is not in the document]
Comment (zevorn) [reply src/data-root.ts:250]: Fixed in cc41587. Web migration now returns immediately when its higher-priority flattened source cannot be copied, so legacy-default imports cannot occupy the skipped path. A Windows junction regression verifies the lower-priority source stays deferred and the preferred source wins after retry.

**Comment by zevorn** [comment]
@codex review

File: src/data-root.ts [hunk 2]
```diff
@@ -288,13 +308,17 @@ export function migrateLegacyWebState(input: {
     const copiedLegacyDshHome = !hasLegacyDshHome
       || copyDirectoryContents(legacyDshHome, input.dataRoot)
     let foundLegacyState = hasLegacyDshHome
+    let copiedSharedEntries = true
     for (const entry of WEB_SHARED_ENTRIES) {
```
[the remark this answers is not in the document]
Comment (zevorn) [reply src/data-root.ts:335]: Fixed in 5407900. An incomplete legacy-default DSH copy now returns before top-level preferences are imported, preserving source priority until the failed link can be retried. A Windows junction regression verifies that the preferred nested path wins after recovery.

**Comment by zevorn** [comment]
@codex review

File: src/data-root.ts [hunk 3]
```diff
@@ -145,41 +145,47 @@ function copyEntry(
...
       copyFileSync(source, destination, fsConstants.COPYFILE_EXCL)
     } catch (error) {
       if ((error as NodeJS.ErrnoException).code !== 'EEXIST') throw error
     }
-    return
+    return true
   }
 
-  if (!sourceStat.isSymbolicLink() || destinationStat !== undefined) return
+  if (!sourceStat.isSymbolicLink() || destinationStat !== undefined) return true
```
[the remark this answers is not in the document]
Comment (zevorn) [reply src/data-root.ts:199]: Fixed in eebd5e8. Migration now reports completion separately from whether it copied state. Web and Desktop both stop initialization when a retryable copy is incomplete, so profile/session directories cannot be created and mistaken for target-wins state on the next launch. Added a Windows regression test proving Web profile setup and runtime construction remain deferred. Full test suite and production build pass.

**Comment by zevorn** [comment]
@codex review
